Runtime/Tracking/Collision/CollisionIgnorer.cs

```diff
@@ -0,0 +1,213 @@
+﻿namespace Zinnia.Tracking.Collision
+{
+    using UnityEngine;
+    using System.Collections.Generic;
+    using Malimbe.MemberChangeMethod;
+    using Malimbe.XmlDocumentationAttribute;
+    using Malimbe.PropertySerializationAttribute;
+    using Zinnia.Data.Collection;
+
+    /// <summary>
+    /// Ignores the collisions between the source <see cref="GameObject"/> colliders and the target <see cref="GameObject"/> colliders.
+    /// </summary>
+    public class CollisionIgnorer : MonoBehaviour
+    {
+        /// <summary>
+        /// The sources to ignore colliders from.
+        /// </summary>
+        [Serialized]
+        [field: DocumentedByXml]
+        public GameObjectObservableList Sources { get; set; }
+        /// <summary>
+        /// The targets to ignore colliders with.
+        /// </summary>
+        [Serialized]
+        [field: DocumentedByXml]
+        public GameObjectObservableList Targets { get; set; }
+
+        protected virtual void OnEnable()
+        {
+            RegisterSourceListeners();
+            RegisterTargetListeners();
+            ToggleCollisions(true);
+        }
+
+        protected virtual void OnDisable()
+        {
+            UnregisterSourceListeners();
+            UnregisterTargetListeners();
+            ToggleCollisions(false);
+        }
+
+        /// <summary>
+        /// Registers the listeners for elements that are added or removed from <see cref="Sources"/>.
+        /// </summary>
+        protected virtual void RegisterSourceListeners()
+        {
+            Sources.ElementAdded.AddListener(AddSource);
```

Comment:
```suggestion
            Sources.ElementAdded.AddListener(OnSourceAdded);
```
I don't like the verb name for that method. It's a callback, not doing an add operation.

**Same for the Remove callback methods for both the Source and Target.** The XML documentation comments also need to change on those methods to explain they're callbacks and actually don't "do that operation".

Reply:
I wouldn't say these are simply callbacks. They are actually carrying out a function. You could call one of these methods directly (for instance in an inherited class) that passes a GameObject into that method.

On that, i'd say the AddSource is still probably correct naming.

Reply:
It's not adding a source, though. Perhaps `IgnoreSource` tells what it does better, but I'm not sure about the equivalent for `Remove` then. `UnignoreSource` seems strange 😄

Reply:
`AddSource` is actually ignoring the given gameobejct against the Targets list.

I guess anyone that wants to replicate the result of the method can just directly call the `ToggleCollisions` method and these can be purely used for the listeners so rename them to `OnSourceAdded` etc.